libevm/VM.cpp

```diff
@@ -643,7 +657,296 @@ void VM::interpretCases()
+			updateIOGas();
+
+			xdiv(m_code[++m_PC]); ++m_PC;
+		}
+		CONTINUE
+         
+		CASE(XSDIV)
+		{
+			ON_OP();
+			updateIOGas();
+
+			xsdiv(m_code[++m_PC]); ++m_PC;
+		}
+		CONTINUE
+        
+		CASE(XMOD)
+		{
+			ON_OP();
+			updateIOGas();
+
+			xmod(m_code[++m_PC]); ++m_PC;
+		}
+		CONTINUE
+         
+		CASE(XSMOD)
+		{
+			ON_OP();
+			updateIOGas();
+
+			xsmod(m_code[++m_PC]); ++m_PC;
+		}
+		CONTINUE
+        
+		CASE(XLT)
+		{
+			ON_OP();
+			updateIOGas();
+
+			xlt(m_code[++m_PC]); ++m_PC;
+		}
+		CONTINUE
+          
+		CASE(XGT)
+		{
+			ON_OP();
+			updateIOGas();
+
+			xgt(m_code[++m_PC]); ++m_PC;
+		}
+		CONTINUE
+          
+		CASE(XSLT)
+		{
+			ON_OP();
+			updateIOGas();
+
+			xslt(m_code[++m_PC]); ++m_PC;
+		}
+		CONTINUE
+         
+		CASE(XSGT)
+		{
+			ON_OP();
+			updateIOGas();
+
+			xsgt(m_code[++m_PC]); ++m_PC;
+		}
+		CONTINUE
+         
+		CASE(XEQ)
+		{
+			ON_OP();
+			updateIOGas();
+
+			xeq(m_code[++m_PC]); ++m_PC;
+		}
+		CONTINUE
+          
+		CASE(XISZERO)
+		{
+			ON_OP();
+			updateIOGas();
+
+			xzero(m_code[++m_PC]); ++m_PC;
+		}
+		CONTINUE
+      
+		CASE(XAND)
+		{
+			ON_OP();
+			updateIOGas();
+
+			xand(m_code[++m_PC]); ++m_PC;
+		}
+		CONTINUE
+         
+		CASE(XXOR)
+		{
+			ON_OP();
+			updateIOGas();
+
+			xoor(m_code[++m_PC]); ++m_PC;
+		}
+		CONTINUE
+         
+		CASE(XNOT)
+		{
+			ON_OP();
+			updateIOGas();
+
+			xnot(m_code[++m_PC]); ++m_PC;
+		}
+		CONTINUE
+         
+		CASE(XSHL)
+		{
+			ON_OP();
+			updateIOGas();
+
+			xshl(m_code[++m_PC]); ++m_PC;
+		}
+		CONTINUE
+         
+		CASE(XSHR)
+		{
+			ON_OP();
+			updateIOGas();
+
+			xshr(m_code[++m_PC]); ++m_PC;
+		}
+		CONTINUE
+         
+		CASE(XSAR)
+		{
+			ON_OP();
+			updateIOGas();
+
+			xsar(m_code[++m_PC]); ++m_PC;
+		}
+		CONTINUE
+         
+		CASE(XROL)
+		{
+			ON_OP();
+			updateIOGas();
+
+			xrol(m_code[++m_PC]); ++m_PC;
+		}
+		CONTINUE
+         
+		CASE(XROR)
+		{
+			ON_OP();
+			updateIOGas();
+
+			xror(m_code[++m_PC]); ++m_PC;
+		}
+		CONTINUE
+
+		CASE(XMLOAD)
+		{
+			updateMem(toInt63(m_SP[0]) + 32);
+			ON_OP();
+			updateIOGas();
+
+			xmload(m_code[++m_PC]); ++m_PC;
+		}
+		CONTINUE
+
+		CASE(XMSTORE)
+		{
+			updateMem(toInt63(m_SP[0]) + 32);
+			ON_OP();
+			updateIOGas();
+
+			xmstore(m_code[++m_PC]); ++m_PC;
+		}
+		CONTINUE
+
+		CASE(XSLOAD)
+		{
+			m_runGas = toInt63(m_schedule->sloadGas);
+			ON_OP();
+			updateIOGas();
+
+			xsload(m_code[++m_PC]); ++m_PC;
+		}
+		CONTINUE
+
+		CASE(XSSTORE)
+		{
+			sstoreGas();
+			ON_OP();
+			updateIOGas();
+	
+			xsstore(m_code[++m_PC]); ++m_PC;
+		}
+		CONTINUE
+
+		CASE(XVTOWIDE)
+		{
+			ON_OP();
+			updateIOGas();
+
+			xvtowide(m_code[++m_PC]); ++m_PC;
+		}
+		CONTINUE
+
+		CASE(XWIDETOV)
+		{
+			ON_OP();
+			updateIOGas();
+
+			xwidetov(m_code[++m_PC]); ++m_PC;
+		}
+		CONTINUE
+
+		CASE(XPUSH)
+		{
+			ON_OP();
+			updateIOGas();
+
+			xpush(m_code[++m_PC]); ++m_PC;
+		}
+		CONTINUE
+
+		CASE(XPUT)
+		{
+			uint8_t b = ++m_PC;
```

Comment:
Oh - ON_OP() just because I forgot.  The rest because they have complicated gas functions I haven't written yet.